pkg/download/download.go

```diff
@@ -163,9 +180,32 @@ func createConfigsFromSingleConfigurationAPI(
 		return err
 	}
 
+	apiId := api.GetId()
+
+	err = ycreator.ReadYamlFile(fs, subPath, apiId)
+	if err != nil {
+		util.Log.Debug("No previously downloaded configs found.")
+	}
+
 	idVal := api.NewIdValue()
 
-	name, cleanName, filter, err := jcreator.CreateJSONConfig(fs, client, api, idVal, subPath)
+	// At this point all API specific substitutions have been made (e.g. reports name = dashboard id)
+	configId, err := getConfigId(idVal.Id, idVal.Name, api)
```

Comment:
It took me way to long to realize in the diff view that this is in the single configuration API method, where you will always only have one config. 
Please just use and sanitize the API ID directly - creating the 'blank' ID Value and parsing it like a real value is just useless operations - and with the incorrect copy pasted comment it's confusing/will possibly make others think there's an error in here in the future.

~~I don't understand what this does/is meant to do tbh.~~

~~`api.NewIdValue()` as I read it will create a struct containing the api's ID as Id and Name value. (for some reason)
Your getConfigId will sanitize this down to a string usable as a filename on all OSs - in either case sanitizing the API ID though. So e.g. this will be `dasboard-v2`~~

~~You comment makes me think that the content of idVal is expected to be something else than what api.NewIdValue() returns? :confused:~~